pyHalo/Halos/HaloModels/TNFWFromParams.py

```diff
@@ -0,0 +1,146 @@
+from pyHalo.Halos.halo_base import Halo
+from lenstronomy.LensModel.Profiles.tnfw import TNFW as TNFWLenstronomy
+import numpy as np
+from pyHalo.Halos.tnfw_halo_util import tnfw_mass_fraction
+from lenstronomy.LensModel.Profiles.tnfw import TNFW
+
+class TNFWFromParams(Halo):
+
+
+    KEY_RT = "r_trunc_kpc"
+    KEY_RS = "rs"
+    KEY_RHO_S = "rho_s"
+    KEY_RV = "rv"
+
+    mass_bound_uss_massfraction = True 
+
+    """
+    The base class for a truncated NFW halo
+    """
+    def __init__(self, mass, x_kpc, y_kpc, r3d, z,
+                 sub_flag, lens_cosmo_instance, args, unique_tag=None):
+        """
+        Denfines a TNFW subhalo with physical params r_trunc_kpc, rs, rhos passed in the args argument
+        """
+
+        self._lens_cosmo = lens_cosmo_instance
+
+        self._kpc_per_arcsec_at_z = self._lens_cosmo.cosmo.kpc_proper_per_asec(z)
+
+        x = x_kpc / self._kpc_per_arcsec_at_z
+
+        y = y_kpc / self._kpc_per_arcsec_at_z
+
+        self._params_physical = args
+
+        mdef = 'TNFW'
+
+        super(TNFWFromParams, self).__init__(mass, x, y, r3d, mdef, z, sub_flag,
+                                           lens_cosmo_instance, args, unique_tag)
+
+    def density_profile_3d(self, r, params_physical=None):
+        """
+        Computes the 3-D density profile of the halo
+        :param r: distance from center of halo [kpc]
+        :return: the density profile in units M_sun / kpc^3
+        """
+        
+        _params = self._params_physical if params_physical is None else params_physical
+
+        r_t = _params[self.KEY_RT]
+        r_s = _params[self.KEY_RS]
+        rho_s = _params[self.KEY_RHO_S]
+
+        n = 1
+
+
+        x = r / r_s
+        tau = r_t / r_s
+
+        n = 0
```

Comment:
why n=0?

Reply:
Good catch, this was left in from when I was running some checks. Will remove.

Reply:
n should be 1